src/_base/harness/scripts/xdebug_version_sync.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+PREPARE_CONFIG=0
+RESTART_FPM=0
+
+xdebug_version_sync()
+{
+  INSTALLED_VERSION=$(docker-compose exec -T -u root "$1" pecl info xdebug 2>&1 | grep "Release Version" | awk '{print $3}')
+  INSTALLED_MAJOR_VERSION=$(echo "$INSTALLED_VERSION" | cut -d '.' -f1)
+
+  if [ "$INSTALLED_MAJOR_VERSION" == "$XDEBUG_VERSION" ]; then
+    echo "Xdebug in $1 is already on the desired major version $XDEBUG_VERSION (found installed version $INSTALLED_VERSION)"
+    return 0
+  fi
+
+  case $XDEBUG_VERSION in
+    "3")
+      INSTALL_CANDIDATE="xdebug"
+      ;;
+    "2")
+      INSTALL_CANDIDATE="xdebug-2.9.8"
+      ;;
+    *)
+      echo "Invalid Xdebug version provided - please check the 'php.ext-xdebug.version' attribute is '2' or  '3'."
+      exit 1
+  esac
+
+  run docker-compose exec -T -u root "$1" pecl uninstall xdebug
+  run docker-compose exec -T -u root "$1" pecl install "$INSTALL_CANDIDATE"
+
+  PREPARE_CONFIG=1
+
+  if [ "$1" == "php-fpm" ]; then
+    RESTART_FPM=1
+  fi
+}
+
+xdebug_version_sync 'console'
+xdebug_version_sync 'php-fpm'
+
+if [ "$PREPARE_CONFIG" == 1 ]; then
+  run ws install --step=prepare
+fi
+
+if [ "$RESTART_FPM" == 1 ]; then
+  run ws service php-fpm restart
```

Comment:
[`ws service php-fpm restart`](https://github.com/inviqa/harness-base-php/blob/0.12.x/src/_base/harness/config/commands.yml#L163-L171) also ensures the rendered configuration is copied into the console and php-fpm containers, so would need to run in the case of an upgrade of Xdebug in the console container to get it using the new Xdebug v3 configuration.

Reply:
Ah good catch! I've combined the two variables into just `UPDATE_CONTAINERS` since we would need to run the FPM restart in both cases, and I see the restart command already does `ws install --step=prepare` as part of it.